iso15118/secc/states/iso15118_2_states.py

```diff
@@ -2338,15 +2305,16 @@ async def process_message(
             )
         )
         if isinstance(present_current, PVEVSEPresentCurrent):
-            present_current_in_a = (
-                present_current.value * 10**present_current.multiplier
-            )
-            target_current = precharge_req.ev_target_current
-            target_current_in_a = target_current.value * 10**target_current.multiplier
+            present_current_in_a = present_current.get_decimal_value()
+            target_current_in_a = ev_data_context.target_current
         else:
-            present_current_in_a = present_current.value
-            target_current = precharge_req.ev_target_current
-            target_current_in_a = target_current.value
+            self.stop_state_machine(
+                f"Error reading EVSE Present Current.
+                Wrong type: {type(present_current)}",
```

Comment:
```suggestion
                f"Error reading EVSE Present Current.",
                f"Wrong type: {type(present_current)}",
```

Reply:
it was updated, closing for a fresh review